packages/headless/src/features/generated-answer/generated-answer-actions.ts

```diff
@@ -0,0 +1,125 @@
+import {
+  ArrayValue,
+  BooleanValue,
+  NumberValue,
+  RecordValue,
+  StringValue,
+} from '@coveo/bueno';
+import {createAction, createAsyncThunk} from '@reduxjs/toolkit';
+import {EventSourcePolyfill} from 'event-source-polyfill';
+import {AsyncThunkGeneratedAnswerOptions} from '../../api/generated-answer/generated-answer-client';
+import {
+  GeneratedAnswerCitationsPayload,
+  GeneratedAnswerMessagePayload,
+} from '../../api/generated-answer/generated-answer-event-payload';
+import {
+  ConfigurationSection,
+  GeneratedAnswerSection,
+  SearchSection,
+} from '../../state/state-sections';
+import {validatePayload} from '../../utils/validate-payload';
+import {buildStreamingRequest} from './generated-awswer-request';
+
+type StateNeededByGeneratedAnswerStream = ConfigurationSection &
+  SearchSection &
+  GeneratedAnswerSection;
+
+const stringValue = new StringValue({required: true});
+const optionalStringValue = new StringValue();
+const booleanValue = new BooleanValue({required: true});
+const citationSchema = {
+  id: stringValue,
+  title: stringValue,
+  uri: stringValue,
+  permanentid: stringValue,
+  clickUri: optionalStringValue,
+};
+
+export interface SSEErrorPayload {
```

Comment:
What's a SSE? :D

Reply:
Server Sent Event. I could call it `GeneratedAnswerStreamErrorPayload` instead